build, win: faster Release rebuilds

bzoz opened the pull request:
Sets Link Time Code Generation to `INCREMENTAL`, improving Release rebuilds speed. The `/LTCG:INCREMENTAL` was added in VS2015 and is not directly supported in `gyp`, thus it is implemented with `AdditionalOptions`. Some more information can be found in this [vcblog post](https://blogs.msdn.microsoft.com/vcblog/2014/11/12/speeding-up-the-incremental-developer-build-scenario/). 

Adds `exe-only` option to `vcbuild.bat`, which will build only `node.exe` binary. It will skip building `cctest`, which takes as long as building `node.exe` itself.

Currently, on my box `vcbuild` rebuild takes 8 minutes. With those changes `vcbuild exe-only` takes only 1.5 minute.

Ref: https://github.com/nodejs/node/issues/17253
<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)
build

/cc @nodejs/build @nodejs/platform-windows

bzoz commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/11806/

seishun commented on common.gypi:
```diff
@@ -186,10 +186,12 @@
             ],
           },
           'VCLinkerTool': {
-            'LinkTimeCodeGeneration': 1, # link-time code generation
```
Why is this removed?

bzoz replied:
It adds `/LTCG` to the linker options, we want to use `/LTCG:INCREMENTAL`. Right now `gyp` does not support this option directly, so we add it through additional options.

seishun commented on vcbuild.bat:
```diff
@@ -51,6 +51,7 @@ set "common_test_suites=%js_test_suites% doctool addons addons-napi&set build_ad
 set http2_debug=
 set nghttp2_debug=
 set link_module=
+set exe_only=
```
That's inconsistent with other arguments which looks like `<feature>` or `no<feature>`. How about `nocctest` instead? Or even better - maybe build `cctest` only if the `test` argument is specified?

seishun commented:
Is there any difference in benchmark results?

bzoz commented:
Updated, changed `exe-only` to `no-cctest`

I did run benchmarks over the weekend, comparing node built with vs2017 - "old" using `/LTCG` and "new" using `/LTCG:INCREMENTAL`. Here are gists with results:
- https://gist.github.com/bzoz/f3055c3b67ff8b33d4091e662f361bec
- https://gist.github.com/bzoz/369da20c8f4cf0d87b29dbf440df62f3
- https://gist.github.com/bzoz/57516ac85213543a673e67a8eced449d
- https://gist.github.com/bzoz/05ce08e671a430e7231014204ed8f739
- https://gist.github.com/bzoz/08ae6c5c2ae67ccab0fae0fd38773d9b

They seem to pretty much the same, although I'm not an expert in this.

bzoz commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/11941/

bzoz commented:
Landed in https://github.com/nodejs/node/commit/8514ea9282d3ecdad487d34aed10b89d8d8d0fbf

MylesBorins commented:
@nodejs/platform-windows is this something we want to backport to 8.x?

Seems like windows tooling for 8.x in general has fallen behind

refack commented:
I'd say backport if it lands smoothly. It's mostly a dev time optimization, IMHO the LTS line can take it or leave it...
